realm/src/main/java/io/realm/RealmBaseAdapter.java

```diff
@@ -89,7 +94,20 @@ public long getItemId(int i) {
      * @param realmResults the new RealmResults coming from the new query.
      */
     public void updateRealmResults(RealmResults<T> realmResults) {
+        if (listener != null) {
```

Comment:
This section could use a comment along the lines of "Making sure that Adapter is refreshed correctly if new RealmResults come from another Realm"